opensearch-project/security
# Authorize rest requests

### Description
Continuation of #2601. Adds support for legacy naming conventions
* Category : Enhancement
* Why these changes are required? To support legacy action names to authorize REST requests

### Issues Resolved
https://github.com/opensearch-project/security/issues/2752


### Testing
Manual testing.

1. Setup opensearch-cluster and helloWorld extension following this [developer guide](https://github.com/opensearch-project/opensearch-sdk-java/blob/main/DEVELOPER_GUIDE.md).
2. Check-out this branch.
3. Install security in opensearch.
4. Ensure certificates are setup correctly in config folder and in `opensearch.yml`
5. Following should be the output of each curl command:
```console
➜  ~ curl -XGET https://new-user:admin@localhost:9200/_extensions/_hw/hello --insecure
Hello, World!%
➜  ~ curl -XPUT https://new-user:admin@localhost:9200/_extensions/_hw/hello/HW --insecure -H 'Content-Type: application/json' -d '{}'
Updated the world's name to HW%
➜  ~ curl -XGET https://new-user:admin@localhost:9200/_extensions/_hw/hello --insecure
Hello, HW!%
➜  ~ curl -XPOST https://new-user:admin@localhost:9200/_extensions/_hw/hello --insecure -H 'Content-Type: application/json' -d '{ }'
no permissions for [hw:greet_with_adjective] and User [name=new-user, backend_roles=[], requestedTenant=null]%
➜  ~ curl -XDELETE https://new-user:admin@localhost:9200/_extensions/_hw/goodbye --insecure
no permissions for [hw:goodbye] and User [name=new-user, backend_roles=[], requestedTenant=null]%
```


### Check List

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
